test/exit-format-sol.test.ts

```diff
@@ -141,4 +146,199 @@ describe("ExitFormat (solidity)", function () {
       BigNumber.from(amount).mul(2)
     );
   });
+
+  it("Can execute a single ERC20 asset exit", async function () {
+    const [alice] = await ethers.getSigners();
+
+    // Alice gets all of the initial minting of tokens
+    let initialSupply = ethers.utils.parseEther((1000).toString());
+    let erc20Token = await deployERC20(alice, initialSupply);
+
+    // Alice transfers all tokens to the TestConsumer
+    await erc20Token
+      .connect(alice)
+      .transfer(testConsumer.address, initialSupply);
+    expect(await erc20Token.balanceOf(alice.address)).to.equal(0);
+    expect(await erc20Token.balanceOf(testConsumer.address)).to.equal(
+      initialSupply
+    );
+
+    // an exit referring to the token contract
+    const singleAssetExit: SingleAssetExit = makeSimpleExit({
+      asset: erc20Token.address,
+      destination: alice.address,
+      amount: initialSupply,
+      tokenMetadata: {
+        assetType: AssetType.ERC20,
+        metadata: "0x",
+      },
+    });
+
+    // Use the exit to withdraw the tokens
+    await (await testConsumer.executeSingleAssetExit(singleAssetExit)).wait();
+    expect(await erc20Token.balanceOf(alice.address)).to.equal(initialSupply);
+    expect(await erc20Token.balanceOf(testConsumer.address)).to.equal(0);
+  });
+
+  it("Can execute a single ERC721 asset exit", async function () {
+    const [alice] = await ethers.getSigners();
+    const tokenId = 11;
+
+    // Alice gets all of the initial minting of tokens
+    let erc721Collection = await deployERC721(alice);
+
+    // Alice transfers all tokens to the TestConsumer
+    await erc721Collection.transferFrom(
+      alice.address,
+      testConsumer.address,
+      tokenId
+    );
+    expect(await erc721Collection.ownerOf(tokenId)).to.equal(
+      testConsumer.address
+    );
+
+    // an exit referring to the token contract
+    const singleAssetExit: SingleAssetExit = makeSimpleExit({
+      asset: erc721Collection.address,
+      destination: alice.address,
+      amount: 1,
+      tokenMetadata: {
+        assetType: AssetType.ERC721,
+        metadata: makeTokenIdExitMetadata(tokenId),
+      },
+    });
+
+    // Use the exit to withdraw the tokens
+    await (await testConsumer.executeSingleAssetExit(singleAssetExit)).wait();
+    expect(await erc721Collection.ownerOf(tokenId)).to.equal(alice.address);
+  });
+
+  it("ERC721 exits with amount != 1 fail", async function () {
+    const [alice] = await ethers.getSigners();
+    let erc721Collection = await deployERC721(alice);
+    const tokenId = 11;
```

Comment:
Be nice to see a test with an invalid tokenId.

Reply:
Added